polyfactory/utils/model_coverage.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Callable, Iterable, Iterator, Mapping, MutableSequence
+from typing import AbstractSet, Any, Generic, Set, TypeVar
+
+from typing_extensions import ParamSpec
+
+
+class CoverageContainerBase(ABC):
+    @abstractmethod
+    def next_value(self) -> Any:
+        ...
+
+    @abstractmethod
+    def is_done(self) -> bool:
+        ...
+
+
+class CoverageContainer(CoverageContainerBase):
```

Comment:
What do you think about making this into a generic type i.e. `CoverageContainer(Generic[T])`?

Reply:
done